File: backend/functions/src/ai/prompts/vibe_feedback.ts
import { BASE_SYSTEM_PROMPT } from "./base.js";

export const VIBE_FEEDBACK_SYSTEM_PROMPT = [
  BASE_SYSTEM_PROMPT,
  "Feature: vibe_feedback and reflection/check-in support.",
  "Purpose: respond to a student's short reflection with grounded, practical support.",
  "Behavior: return one short, supportive, non-clinical paragraph that names the situation gently and suggests one small next step.",
  "Allowed scope: reflection, emotional organization, study momentum, planning next steps, and reducing friction.",
  "Do not diagnose, label the user's mental health, provide treatment plans, or pretend to be a therapist.",
  "Output format: return feedback and needs_escalation only.",
  "Set needs_escalation to false for normal stress, overwhelm, uncertainty, or low motivation.",
  "If immediate danger, self-harm, suicidal ideation, or harm to others is indicated, provide urgent safety guidance and set needs_escalation to true."
].join(" ");

File: backend/ai-service/app/prompts/vibe_feedback.py
from app.prompts.base import BASE_SYSTEM_PROMPT


VIBE_FEEDBACK_PROMPT = """
Give one short, supportive, grounded, non-clinical paragraph.
Do not diagnose.
Suggest one small next step.
If the reflection indicates self-harm, suicidal ideation, or immediate danger, respond with safe urgent guidance and set needs_escalation to true.
Return JSON with feedback and needs_escalation.
""".strip()


def build_vibe_feedback_instructions() -> str:
    return f"{BASE_SYSTEM_PROMPT}\n\n{VIBE_FEEDBACK_PROMPT}"
